utils/logalloc.cc

```diff
@@ -1327,7 +1338,7 @@ segment_pool::allocate_or_fallback_to_reserve() {
     auto seg = allocate_segment(_current_emergency_reserve_goal);
     if (!seg) {
         _allocation_failure_flag = true;
-        throw std::bad_alloc();
+        throw bad_alloc(format("failed to allocate segment (_current_emergency_reserve_goal={})", _current_emergency_reserve_goal));
```

Comment:
There's a chance we'll fail to allocate during formatting, but it will just throw the original bad_alloc, so no problem.